pkg/eval/value_test.go

```diff
@@ -16,6 +16,7 @@ var reprTests = []struct {
 	{"a\nb", `"a\nb"`},
 	{"foo bar", "'foo bar'"},
 	{"a\x00b", `"a\x00b"`},
+	{"a\033b", `"a\eb"`},
```

Comment:
Not sure what this additional test case is supposed to do either.

Reply:
It reflects the fact that in the past the `\e` sequence wasn't recognized. I complained that even though Go doesn't recognize it the `\e` sequence is useful in programs written in Elvish and POSIX shells. Somewhere along the line you introduced support for `\e` but didn't document it or, AFAICT, include a test that it is emitted when encoding an Esc char in a string. This test addresses the second point.